src/dotty/tools/dotc/repl/CompilingInterpreter.scala

```diff
@@ -662,29 +663,71 @@ class CompilingInterpreter(
 
     private class GenericHandler(statement: Tree) extends StatementHandler(statement)
 
-    private class ValHandler(statement: ValDef) extends StatementHandler(statement) {
-      override val boundNames = List(statement.name)
+    private abstract class ValOrPatHandler(statement: Tree)
+        extends StatementHandler(statement) {
+      override val boundNames: List[Name] = _boundNames
       override def valAndVarNames = boundNames
 
       override def resultExtractionCode(req: Request, code: PrintWriter): Unit = {
-        val vname = statement.name
-        if (!statement.mods.is(Flags.AccessFlags) &&
-          !(isGeneratedVarName(vname.toString) &&
-            req.typeOf(vname.encode) == "Unit")) {
-          val prettyName = vname.decode
-          code.print(" + \"" + prettyName + ": " +
-            string2code(req.typeOf(vname)) +
-            " = \" + " +
-            " (if(" +
-            req.fullPath(vname) +
-            ".asInstanceOf[AnyRef] != null) " +
-            " ((if(" +
-            req.fullPath(vname) +
-            ".toString().contains('\\n')) " +
-            " \"\\n\" else \"\") + " +
-            req.fullPath(vname) + ".toString() + \"\\n\") else \"null\\n\") ")
-        }
+        if (!shouldShowResult(req)) return
+
+        val resultExtractors =
+          for (varName <- boundNames)
+            yield resultExtractionCode(req, varName)
+        code.print(resultExtractors.mkString(""))
+      }
+
+      private def resultExtractionCode(req: Request, varName: Name): String = {
+        def if_(condition: String)(thenBranch: String)(elseBranch: String): String =
+          s"(if ($condition) {$thenBranch} else {$elseBranch})"
+
+        val prettyName = varName.decode
+        val varType = string2code(req.typeOf(varName))
+        val fullPath = req.fullPath(varName)
+
+        s""" + "$prettyName: $varType = " + """ +
+          if_(s"$fullPath.asInstanceOf[AnyRef] != null") {
+            if_(s"$fullPath.toString().contains('\\n')") { "\"\\n\"" }
+            /* else */ { "\"\"" } +
+              s""" + $fullPath.toString() + "\\n" """
+          } /* else */ {
+            "\"null\\n\""
+          }
       }
```

Comment:
I find there are too many levels of escaping to be able to understand it. Can we try to put everything between
one pair of """s and use only interpolation inside? That should also obviate the need for `if_`.